types/styled-components/index.d.ts

```diff
@@ -124,7 +124,7 @@ type ReactDefaultProps<C> = C extends { defaultProps: infer D } ? D : never;
 export type AnyStyledComponent = StyledComponent<any, any, any, any> | StyledComponent<any, any, any>;
 
 export type StyledComponent<
-    C extends keyof JSX.IntrinsicElements | React.ComponentType<any>,
+    C extends keyof React.JSX.IntrinsicElements | React.ComponentType<any>,
```

Comment:
```suggestion
    C extends React.ElementType,
```

Reply:
@eps1lon wouldn't this be the same thing?

Reply:
I believe so, yes.